packages/build-tools/src/steps/functionGroups/submit.ts

```diff
@@ -0,0 +1,732 @@
+import { findArtifacts } from '../../utils/artifacts';
+import { createCheckoutBuildFunction } from '../functions/checkout';
+import { createSubmissionEntityFunction } from '../functions/createSubmissionEntity';
+import { createDownloadArtifactFunction } from '../functions/download_artifact';
+import { resolveIosArtifactPath } from '../functions/findAndUploadBuildArtifacts';
+import { createInstallNodeModulesBuildFunction } from '../functions/installNodeModules';
+import { createSetUpNpmrcBuildFunction } from '../functions/useNpmToken';
+
+export function createEasSubmitBuildFunctionGroup(
+  buildToolsContext: CustomBuildContext<BuildJob>
+): BuildFunctionGroup {
+  return new BuildFunctionGroup({
+    namespace: 'eas',
+    id: 'submit',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'application_archive_path',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'application_archive_url',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'build_id',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'profile',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'is_verbose_fastlane_enabled',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.BOOLEAN,
+      }),
+    ],
+    createBuildStepsFromFunctionGroupCall: (globalCtx, { inputs }) => {
+      validateInputs(inputs);
+
+      const buildId = String(inputs.build_id.value ?? globalCtx.env.EAS_BUILD_ID);
+      const profile = inputs.profile.value ? String(inputs.profile.value) : undefined;
+      const isVerboseFastlaneEnabled = inputs.is_verbose_fastlane_enabled.value
+        ? String(inputs.is_verbose_fastlane_enabled.value)
+        : undefined;
+
+      const applicationArchivePath = inputs.application_archive_path.value
+        ? String(inputs.application_archive_path.value)
+        : undefined;
+
+      const applicationArchiveUrl = inputs.application_archive_url.value
+        ? String(inputs.application_archive_url.value)
+        : undefined;
+
+      const commandOptions = {
+        buildId,
+        profile,
+        isVerboseFastlaneEnabled,
+        applicationArchivePath,
+        applicationArchiveUrl,
+      };
+
+      if (buildToolsContext.job.platform === Platform.IOS) {
+        return createStepsForIosSubmit(commandOptions, globalCtx);
+      } else {
+        return createStepsForAndroidSubmit(commandOptions, globalCtx);
+      }
+    },
+  });
+}
+
+function validateInputs(inputs: BuildStepInputById): void {
+  const providedArgs = [
+    inputs.build_id.value,
```

Comment:
Should we make sure this errors if none is provided and there's also no `env.BUILD_ID`?